app/controllers/topics_controller.rb

```diff
@@ -2,4 +2,23 @@ class TopicsController < ApplicationController
   def index
     @topics = Topic.all
   end
+
+  def new
+    @topic = Topic.new
+  end
+
+  def create
+    @topic = Topic.create(topic_params)
+    if @topic.valid?
+      redirect_to topics_path
```

Comment:
You could add a flash message to let the user know that the creation succeeded, but perhaps leave that for a later feature enhancement? What do you think Mr Produce Owner?

Reply:
Weirdly I was going to add a flash message - I wasn't sure, and then read about the way I ended up implemented errors, and I preferred it as it would show all messages themselves.... I think you're right though, later, I'll put some flash messages at some point after the core functionality